stubs/pywin32/win32com/client/dynamic.pyi

```diff
@@ -1,14 +1,72 @@
 from _typeshed import Incomplete
-from typing import Any
+from typing import Any, Protocol, TypeVar, overload
+from typing_extensions import TypeAlias
+
+import _win32typing
+from win32.lib.pywintypes import IIDType
+from win32com.client import build as build
+
+class _DispatchCreateClass(Protocol):
+    def __init__(
+        self,
+        IDispatch: str | PyIDispatchType | _GoodDispatchTypes | PyIUnknownType,
+        olerepr: build.DispatchItem | build.LazyDispatchItem,
+        userName: str | None = ...,
+        UnicodeToString: None = ...,
+        lazydata: Incomplete | None = ...,
+    ): ...
+
+_T = TypeVar("_T", bound=_DispatchCreateClass)
+
+debugging: int
+debugging_attr: int
+LCID: int
+ERRORS_BAD_CONTEXT: Incomplete
+ALL_INVOKE_TYPES: Incomplete
+
+def debug_print(*args) -> None: ...
+def debug_attr_print(*args) -> None: ...
+def MakeMethod(func, inst, cls): ...
+
+PyIDispatchType = _win32typing.PyIDispatch
+PyIUnknownType = _win32typing.PyIUnknown
+
+_GoodDispatchTypes: TypeAlias = tuple[type[str], type[IIDType]]  # noqa: Y047
```

Comment:
Not sure we need this `noqa` -- Y047 is "A private `TypeAlias` should be used at least once in the file in which it is defined." You use `_GoodDispatchTypes` just below on line 38!

```suggestion
_GoodDispatchTypes: TypeAlias = tuple[type[str], type[IIDType]]
```

Reply:
Yeah, we already have it as a plugin, but we have that specific check disabled:

https://github.com/python/typeshed/blob/503eab31d7e11fbe7f458a3e96ba9fa1907ce3a3/.flake8#L25-L28

Basically, flake8-pyi's CI is very tightly coupled with `noqa` comments in typeshed. That often requires introducing `noqa` comments into typeshed for flake8-pyi checks that don't exist yet, in order to get flake8-pyi's CI to pass.

It's all slightly unfortunate, but I have yet to think of a better way of doing things :)

Reply:
The problematic check in flake8-pyi's CI is this: https://github.com/PyCQA/flake8-pyi/blob/e8243d7cc648433cbbf3fd0621a13a50db063e80/.github/workflows/check.yml#L100-L117

What would be awesome would be if we could have a non-blocking mypy_primer-esque check at flake8-pyi, instead of a blocking check. Then flake8-pyi's CI wouldn't need to be so tightly coupled with typeshed's `noqa` comments, and we could enable `NQA102` in typeshed's CI. But that sounds like... a lot of work :(

Reply:
Example: #7658 was required to make the CI on https://github.com/PyCQA/flake8-pyi/pull/213 go green. But the CI on #7658 would have failed if we'd had `NQA102` enabled.

Reply:
I see. This makes perfect sense. I think I have an idea, I'll play around. See if I can make an action similar to typeshed's mypy_primer_comment.